src/Microsoft.AspNet.Mvc.Core/RemoteAttribute.cs

```diff
@@ -0,0 +1,237 @@
+            foreach (var field in _additionalFieldsSplit)
+            {
+                delimitedAdditionalFields += "," + FormatPropertyForClientValidation(field);
+            }
+
+            return delimitedAdditionalFields;
+        }
+
+        /// <summary>
+        /// Formats <paramref name="property"/> for use in generated HTML.
+        /// </summary>
+        /// <param name="property">One field name the client should include in a validation request.</param>
+        /// <returns>Name of a field the client should include in a validation request.</returns>
+        /// <remarks>Returns <paramref name="property"/> with a <c>"*."</c> prefix.</remarks>
+        public static string FormatPropertyForClientValidation(string property)
+        {
+            if (string.IsNullOrEmpty(property))
+            {
+                throw new ArgumentException(Resources.ArgumentCannotBeNullOrEmpty, "property");
+            }
+
+            return "*." + property;
+        }
+
+        /// <summary>
+        /// Returns the URL where the client should send a validation request.
+        /// </summary>
+        /// <param name="context">The <see cref="MvcClientModelValidationContext"/> used to generate the URL.</param>
+        /// <returns>The URL where the client should send a validation request.</returns>
+        protected virtual string GetUrl([NotNull] MvcClientModelValidationContext context)
+        {
+            var url = context.UrlHelper.RouteUrl(
+                routeName: RouteName,
+                values: RouteData,
+                protocol: null,
+                host: null,
+                fragment: null);
+            if (url == null)
+            {
+                throw new InvalidOperationException(Resources.RemoteAttribute_NoUrlFound);
+            }
+
+            return url;
+        }
+
+        /// <inheritdoc />
+        public override string FormatErrorMessage(string name)
+        {
+            return string.Format(CultureInfo.CurrentCulture, ErrorMessageString, name);
+        }
+
+        /// <inheritdoc />
+        public override bool IsValid(object value)
+        {
+            return true;
+        }
+
+        /// <inheritdoc />
+        /// <exception cref="ArgumentException">
+        /// Thrown if provided <paramref name="context"/> is not a <see cref="MvcClientModelValidationContext"/>
+        /// instance.
+        /// </exception>
+        /// <exception cref="InvalidOperationException">
+        /// Thrown if unable to generate a target URL for a validation request.
+        /// </exception>
+        public virtual IEnumerable<ModelClientValidationRule> GetClientValidationRules(
+            [NotNull] ClientModelValidationContext context)
+        {
+            var mvcContext = context as MvcClientModelValidationContext;
+            if (mvcContext == null)
+            {
+                var message = Resources.FormatArgumentUnexpectedType(
```

Comment:
This feels dirty. Couldn't we just to pull the `IUrlHelper` from the context's RequestServices? That's the code we'd expect users to write if they required services when writing custom attributes. Alternatively, stick the property on `ClientValidationContext`.